### PR TITLE
Don't emit code to call main if there is no main

### PR DESCRIPTION
**Pull request by kripken**

### CODE REVIEW

**Comment by curiousdannii** [comment]
You should be able to put these lines behind an ifdef too: https://github.com/kripken/emscripten/blob/137382c98097eee658285b7a25dc8470fa55f332/src/postamble.js#L389-396

File: src/jsifier.js [hunk 1]
```diff
@@ -25,9 +25,11 @@ var proxiedFunctionTable = ["null" /* Reserve index 0 for an undefined function*
 // map: pair(sig, syncOrAsync) -> function body
 var proxiedFunctionInvokers = {};
 
+// used internally. set when there is a main() function.
+var HAS_MAIN = '_main' in IMPLEMENTED_FUNCTIONS;
```
Comment (saschanaz) [inline src/jsifier.js]: This is valid in Python but not in JS 😄

**Comment by saschanaz** [comment]
(That jsifier.js looks very Python-ish, I incorrectly thought they are even invalid 😮)

File: src/postamble.js [hunk 2]
```diff
@@ -163,6 +163,7 @@ dependenciesFulfilled = function runCaller() {
   if (!Module['calledRun']) dependenciesFulfilled = runCaller; // try this again later, after new deps are fulfilled
 }
 
+#if HAS_MAIN
 Module['callMain'] = Module.callMain = function callMain(args) {
```
Comment (saschanaz) [inline src/postamble.js]: `Module['callMain']` and `Module.callMain` are same in JS, can we remove the latter part?

**Comment by kripken** [comment]
Thanks @curiousdannii , adding.

File: src/postamble.js (hunk 2 above)
Comment (kripken) [reply src/postamble.js]: Yes, I'm actually not sure why we have this pattern. I suspect it is very old, and either worked around a closure compiler bug or a bug in our understanding of closure compiler ;) We should remove it across the codebase, leaving only `Module['X']`.

**Pull request merged**